dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueNodeSelectionStrategy.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import com.palantir.conjure.java.client.config.NodeSelectionStrategy;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+import java.util.List;
+
+public enum DialogueNodeSelectionStrategy {
+    PIN_UNTIL_ERROR,
+    PIN_UNTIL_ERROR_WITHOUT_RESHUFFLE,
+    BALANCED,
+    UNKNOWN;
+
+    static List<DialogueNodeSelectionStrategy> fromHeader(String header) {
+        return Splitter.on(",").splitToList(header).stream()
+                .map(DialogueNodeSelectionStrategy::safeValueOf)
+                .collect(ImmutableList.toImmutableList());
```

Comment:
string -> list -> stream -> list.

Should we filter out unknowns here?

There's a [splitToStream](https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/base/Splitter.html#splitToStream-java.lang.CharSequence-) function, or we could write an old fashion for loop ;-)

Reply:
Ahh ya, I think our guava isn't modern enough to get splitToStream. I'll upgrade and update